### ISSUE TITLE
Does Scikit-Learn supports MiniBatch or Batch Gradient Descent for Linear Regression

### ISSUE THREAD

**Issue by Kirushikesh** [ageron/handson-ml2#525]
Hi @ageron ,

According to Chapter4, LinearRegression() in Scikit-Learn uses SVD for finding the model parameters. SGDRegressor() uses SGD algorithm to find the models parameters. Even partial_fit method still causes one weight update per training example in the batch, so this isn't really uses mini-batch or batch gradient descent for parameter learning. Am i correct? and is there any method to use mini-batch with batch_size and batch gradient descent using scikit-learn?

**Comment by ageron** [comment]
Hi @Kirushikesh ,

You're 100% correct. I don't believe there's a way to get true mini-batch gradient descent, or batch-gradient descent using in Scikit-Learn. When using `partial_fit()` with batches, the result looks similar to mini-batch gradient descent, but under the hood it's actually SGD. I will point this out in the book, thanks. 🙏

However, you can use Keras to build a simple Linear Regression model trained using mini-batch GD, like this:

```python
import tensorflow as tf

X = tf.random.normal((1000, 10))  # some random inputs
y = tf.random.uniform((1000, 1))  # some random targets

model = tf.keras.Sequential([tf.keras.layers.Dense(1)])
model.compile(loss="mse", optimizer=tf.keras.optimizers.SGD(learning_rate=0.01))
history = model.fit(X, y, epochs=10, batch_size=32)
```

**Comment by Kirushikesh** [comment]
Oh ok understood now. Thanks for the answer @ageron
